static/app/views/alerts/rules/utils.tsx

```diff
@@ -101,6 +102,14 @@ export function getAlertRuleActionCategory(rule: MetricRule) {
   }
 }
 
-export function shouldUseErrorsDiscoverDataset(query: string, dataset: Dataset) {
-  return dataset === Dataset.ERRORS && query?.includes('is:unresolved');
+export function shouldUseErrorsDiscoverDataset(
+  query: string,
+  dataset: Dataset,
+  organization: Organization
+) {
+  if (!hasDatasetSelector(organization)) {
+    return dataset === Dataset.ERRORS && query?.includes('is:unresolved');
+  }
+
+  return dataset === Dataset.ERRORS;
```

Comment:
I'm going to approve the PR since the rest of it seems fine to me. This is the only thing i'm kinda unsure of. It seems safe, but I don't know what the context around the extra check for `query?.includes('is:unresolved')` in the original code is